### PR TITLE
Add missing "VAT Number" field to customer registration page

### LINKED ISSUE #31326: [Issue] Add missing "VAT Number" field to customer registration page
This issue is automatically created based on existing pull request: magento/magento2#31321: Add missing "VAT Number" field to customer registration page

---------
### Description
This commits makes "VAT Number" field visible when `customer/create_account/vat_frontend_visibility` is enabled.
The code was copied from nearby [address/edit.phtml](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Customer/view/frontend/templates/address/edit.phtml#L94-L108) template.

### Manual testing scenarios
1. Enable `Show VAT Number on Storefront` option at _Stores > Configuration > Customers > Customer Configuration > Create New Account Options > Show VAT Number on Storefront_
2. Navigate to frontend and add some product to the cart.
3. Proceed to Multishipping checkout and select "Create Account" option
4. Address fields are shown but "VAT Number" field is missing.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by vovayatsyuk**
### Description
This commits makes "VAT Number" field visible when `customer/create_account/vat_frontend_visibility` is enabled.
The code was copied from nearby [address/edit.phtml](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Customer/view/frontend/templates/address/edit.phtml#L94-L108) template.

### Manual testing scenarios
1. Enable `Show VAT Number on Storefront` option at _Stores > Configuration > Customers > Customer Configuration > Create New Account Options > Show VAT Number on Storefront_
2. Navigate to frontend and add some product to the cart.
3. Proceed to Multishipping checkout and select "Create Account" option
4. Address fields are shown but "VAT Number" field is missing.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#31326: Add missing "VAT Number" field to customer registration page

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/443

### CODE REVIEW

**Comment by gabrieldagama** [comment]
@magento create issue

**Comment by gabrieldagama** [comment]
@magento run all tests

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @vovayatsyuk, thanks for contributing! We must cover such changes with automated tests, I believe MFTF would suit best in this case. Are you able to add MFTF cover to this PR?

Thanks.

**Comment by engcom-Hotel** [comment]
Hi @vovayatsyuk, I will help with the test.

**Comment by vovayatsyuk** [comment]
Thank you!

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests Functional Tests B2B, Functional Tests CE, Functional Tests EE, Integration Tests, Static Tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, Integration Tests, Static Tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Static Tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Static Tests

**Comment by engcom-Hotel** [comment]
@magento Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

File: app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        <!--Check the VAT Number field-->
+        <seeElement selector="{{StorefrontCustomerAddressSection.vatId}}" stepKey="assertVatIdField"/>
+        <after>
+            <magentoCLI command="config:set customer/create_account/vat_frontend_visibility 0" stepKey="showVatNumberOnStorefrontNo"/>
+            <actionGroup ref="CliCacheFlushActionGroup" stepKey="flushCacheAfter">
+                <argument name="tags" value="config"/>
+            </actionGroup>
+            <deleteData stepKey="deleteCategory" createDataKey="category"/>
+            <deleteData stepKey="deleteProduct1" createDataKey="product1"/>
+        </after>
```
Comment (gabrieldagama) [inline app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml]: Let's keep this section after the `<before>` node, to follow the same pattern as other tests and the docs https://devdocs.magento.com/mftf/docs/test.html

File: app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            <testCaseId value="MC-40016"/>
+            <group value="Multishipment"/>
+        </annotations>
+        <before>
+            <magentoCLI command="config:set customer/create_account/vat_frontend_visibility 1" stepKey="showVatNumberOnStorefrontYes"/>
+            <actionGroup ref="CliCacheFlushActionGroup" stepKey="flushCacheBefore">
+                <argument name="tags" value="config"/>
+            </actionGroup>
+            <createData stepKey="category" entity="SimpleSubCategory"/>
+            <createData stepKey="product1" entity="SimpleProduct">
```
Comment (gabrieldagama) [inline app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml]:
```suggestion
            <createData stepKey="product" entity="SimpleProduct">
```

File: app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest">
+        <annotations>
+            <features value="Multishipment"/>
```
Comment (gabrieldagama) [inline app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml:13]: Isn't it possible to verify this change without the multishipment? Ideally, I believe this test should be on the customer module.

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @engcom-Hotel, thanks for adding test coverage, please have a look at my comments below.

File: app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            <description value="'VAT Number' field should be available at create account page  if 'Show VAT Number on Storefront' is Yes"/>
+            <severity value="MAJOR"/>
+            <testCaseId value="MC-40016"/>
+            <group value="Multishipment"/>
+        </annotations>
+        <before>
+            <magentoCLI command="config:set customer/create_account/vat_frontend_visibility 1" stepKey="showVatNumberOnStorefrontYes"/>
+            <actionGroup ref="CliCacheFlushActionGroup" stepKey="flushCacheBefore">
+                <argument name="tags" value="config"/>
+            </actionGroup>
```
Comment (gabrieldagama) [inline app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml]: Do we really need to clear cache?

**Comment by engcom-Hotel** [comment]
@magento run all tests

File: app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml (hunk 4 above)
Comment (engcom-Hotel) [reply app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml]: Has been removed.

File: app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml (hunk 3 above)
Comment (engcom-Hotel) [reply app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml:13]: Has been moved to the customer module.

File: app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml (hunk 2 above)
Comment (engcom-Hotel) [reply app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml]: Has been renamed.

File: app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml (hunk 1 above)
Comment (engcom-Hotel) [reply app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml]: Has been moved according to the pattern.

File: app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml (hunk 3 above)
Comment (engcom-Hotel) [reply app/code/Magento/Multishipping/Test/Mftf/Test/StoreFrontCheckVatIdAtAccountCreateWithMultishipmentTest.xml:13]: Moved back as discussed.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests CE

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests CE

**Review by gabrieldagama — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-8617](https://jira.corp.magento.com/browse/ENGCOM-8617) has been created to process this Pull Request
:eight_spoked_asterisk: @gabrieldagama, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenarios
1. Enable `Show VAT Number on Storefront` option at _Stores > Configuration > Customers > Customer Configuration > Create New Account Options > Show VAT Number on Storefront_
2. Navigate to frontend and add some product to the cart.
3. Proceed to Multishipping checkout and select "Create Account" option

**Before:** :heavy_multiplication_x:  Address fields are shown but "VAT Number" field is missing.

Screenshot from 2021-01-13 11-05-37

**After:** :heavy_check_mark:   "VAT Number" is shown



Also, was checked in different places, with a disabled "Show VAT Number on Storefront" option and everything works as expected.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Pull request merged**